cmake/add_dcap_client_target.cmake

```diff
@@ -0,0 +1,115 @@
+    set(SSLEAY32 ${CMAKE_SOURCE_DIR}/prereqs/nuget/openssl.redist/build/native/bin/v110/x64/Release/dynamic/cdecl/ssleay32.dll)
+    set(ZLIB ${CMAKE_SOURCE_DIR}/prereqs/nuget/zlib.redist/build/native/bin/v110/x64/Release/dynamic/cdecl/zlib.dll)
+
+    # No-op if the DCAP provider is not found
+    if (NOT EXISTS ${AZURE_DCAP_QUOTEPROV})
+        message (WARNING "dcap_quoteprov dependencies were not found, may not execute successfully.")
+    else ()
+        if (NOT EXISTS ${LIBCURL})
+            message(FATAL_ERROR "Found Azure dcap_quoteprov, but could not find its dependency ${LIBCURL}, aborting.")
+        endif ()
+        if (NOT EXISTS ${LIBSSH2})
+            message(FATAL_ERROR "Found Azure dcap_quoteprov, but could not find its dependency ${LIBSSH2}, aborting.")
+        endif ()
+        if (NOT EXISTS ${LIBEAY32})
+            message(FATAL_ERROR "Found Azure dcap_quoteprov, but could not find its dependency ${LIBEAY32}, aborting.")
+        endif ()
+        if (NOT EXISTS ${SSLEAY32})
+            message(FATAL_ERROR "Found Azure dcap_quoteprov, but could not find its dependency ${SSLEAY32}, aborting.")
+        endif ()
+        if (NOT EXISTS ${ZLIB})
+            message(FATAL_ERROR "Found Azure dcap_quoteprov, but could not find its dependency ${ZLIB}, aborting.")
+        endif ()
+
+        # Add copy actions for each of the dependencies
+        add_custom_command(
+            OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/dcap_quoteprov.dll
+            DEPENDS ${AZURE_DCAP_QUOTEPROV}
+            COMMAND ${CMAKE_COMMAND} -E copy ${AZURE_DCAP_QUOTEPROV} ${CMAKE_CURRENT_BINARY_DIR})
+
+        add_custom_command(
+            OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/libcurl.dll
+            DEPENDS ${LIBCURL}
+            COMMAND ${CMAKE_COMMAND} -E copy ${LIBCURL} ${CMAKE_CURRENT_BINARY_DIR})
+            
+        add_custom_command(
+            OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/libssh2.dll
+            DEPENDS ${LIBSSH2}
+            COMMAND ${CMAKE_COMMAND} -E copy ${LIBSSH2} ${CMAKE_CURRENT_BINARY_DIR})
+            
+        add_custom_command(
+            OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/libeay32.dll
+            DEPENDS ${LIBEAY32}
+            COMMAND ${CMAKE_COMMAND} -E copy ${LIBEAY32} ${CMAKE_CURRENT_BINARY_DIR})
+            
+        add_custom_command(
+            OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/ssleay32.dll
+            DEPENDS ${SSLEAY32}
+            COMMAND ${CMAKE_COMMAND} -E copy ${SSLEAY32} ${CMAKE_CURRENT_BINARY_DIR})
+            
+        add_custom_command(
+            OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/zlib.dll
+            DEPENDS ${ZLIB}
+            COMMAND ${CMAKE_COMMAND} -E copy ${ZLIB} ${CMAKE_CURRENT_BINARY_DIR})
+
+        # Add the dependencies to the custom target list of dependencies
+        list(APPEND DEPENDENCIES
+            ${CMAKE_CURRENT_BINARY_DIR}/dcap_quoteprov.dll
```

Comment:
Except for dcap_quoteprov.dll, all other dlls are not needed in version >= 0.0.1 of the Azure DCAP client.